Corona-Warn-App/src/androidTest/java/de/rki/coronawarnapp/eventregistration/storage/TraceLocationCheckInDaoTest.kt

```diff
@@ -82,4 +85,42 @@ class TraceLocationCheckInDaoTest : BaseTestInstrumentation() {
 
         checkInsFlow.first() shouldBe emptyList()
     }
+
+    @Test
+    fun traceLocationCheckInDaoRetrieveById() = runBlocking {
+        val generatedId1 = checkInDao.insert(testCheckIn)
+        val generatedId2 = checkInDao.insert(testCheckIn)
+
+        checkInDao.entryForId(generatedId1)!!.id shouldBe generatedId1
+        checkInDao.entryForId(generatedId2)!!.id shouldBe generatedId2
+    }
+
+    @Test
+    fun traceLocationCheckInDaoDeleteById() = runBlocking {
+        val generatedId1 = checkInDao.insert(testCheckIn)
+        val generatedId2 = checkInDao.insert(testCheckIn)
+
+        checkInDao.deleteByIds(listOf(generatedId1))
+        checkInDao.entryForId(generatedId1) shouldBe null
+        checkInDao.entryForId(generatedId2) shouldNotBe null
+    }
+
+    @Test
+    fun traceLocationCheckInDaoUpdateById() = runBlocking {
+        val generatedId1 = checkInDao.insert(testCheckIn)
+
+        checkInDao.updateEntityById(generatedId1) {
+            it.copy(address = "test")
+        }
+        checkInDao.entryForId(generatedId1)!!.address shouldBe "test"
+    }
+
+    @Test
+    fun traceLocationCheckInDaoUpdateById_raceCondition1(): Unit = runBlocking {
```

Comment:
Why did you call the test raceCondition? Isn't it just asserting that an `IllegalStateException` is thrown when we want to  update a non-existing checkIn?

Reply:
The name was from a previous test case that I adapted :grin:. But I think it still fits and would just keep it as is. Isn't that a racecondition? We try to update something that should exist, but when we try it, it actually doesn't.